packages/x-components/src/components/animations/create-directional-animation-factory.ts

```diff
@@ -19,23 +19,23 @@ export type AnimationOrigin =
  *
  * @internal
  */
-export function createDirectionalAnimationFactory(animationName: string) {
+export function createDirectionalAnimationFactory(
+  animationName: string
+): ReturnType<typeof defineComponent> {
```

Comment:
I don't why, but the return type when using, is resolved to `any`
![Screenshot 2024-08-02 at 1 20 23 PM](https://github.com/user-attachments/assets/3a029028-d614-42a6-b0c0-dbc5758917f9)

```TS
export function createDirectionalAnimationFactory(
  animationName: string
): (animationOrigin: AnimationOrigin) => DefineComponent {
  return (animationOrigin = 'top') => ...
```

That infers better